docs/src/userguide/language_basics.rst

```diff
@@ -760,12 +758,25 @@ An external C++ function that may raise an exception can be declared with::
 
 See :ref:`wrapping-cplusplus` for more details.
 
+Finally, if you are certain that your function cannot raise an exception, (e.g., it
+does not use Python objects at all), you can declare it as such using ``noexcept``::
+
+    cdef int spam() noexcept
+
```

Comment:
```suggestion

If a ``noexcept`` function *does* finish with an exception then it will print a warning message but not allow the exception to propagate further.
```

(i.e. the behaviour is well-defined and safe)